src/sync/app.cpp

```diff
@@ -0,0 +1,77 @@
+#include "app.hpp"
+#include <sstream>
+#include "sync_manager.hpp"
+#include "generic_network_transport.hpp"
+#include "app_credentials.hpp"
+#include "../external/json/json.hpp"
+
+namespace realm {
+
+std::shared_ptr<RealmApp> RealmApp::app(const std::string app_id,
+                                        const realm::util::Optional<RealmApp::Config> config)
+{
+    return std::make_shared<RealmApp>(app_id, config);
+}
+
+void RealmApp::login_with_credentials(const std::shared_ptr<AppCredentials> credentials,
+                                      int timeout,
```

Comment:
Unclear what unit this is, I assume milliseconds, so `timeoutMs` would be more readable.

Reply:
Good catch 👍